### PR TITLE
Waypoint path generation

### PR DESCRIPTION
**Pull request by SPD3**

### CODE REVIEW

File: src/main/java/frc/robot/pathgeneration/waypoints/Waypoint.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  public double getY() {
+    return m_y;
+  }
+
+  @Override
+  public void setLocation(double x, double y) {
+    x = m_x;
+    y = m_y;
```
Comment (KjellSteeman) [inline src/main/java/frc/robot/pathgeneration/waypoints/Waypoint.java:25]: m_x = x are flipped and the same thing with the m_y = y.

File: src/main/java/frc/robot/pathgeneration/pathgenerators/DiagonalPathGenerator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package frc.robot.pathgeneration.pathgenerators;
```
Comment (wpd) [inline src/main/java/frc/robot/pathgeneration/pathgenerators/DiagonalPathGenerator.java:1]: I would like to see more documentation in this file.  What is DiagonalPathGenerator?  Why would I use it?  How do I use it?

File: src/main/java/frc/robot/pathgeneration/waypoints/Waypoint.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package frc.robot.pathgeneration.waypoints;
+
+import java.awt.geom.Point2D;
+
+public class Waypoint extends Point2D {
```
Comment (wpd) [inline src/main/java/frc/robot/pathgeneration/waypoints/Waypoint.java:5]: How is this different and/or better than Point2D?

**Review by KjellSteeman — APPROVED** [review]
We reviewed the code but patrick has some questions but we are going to approve it anyways.

**Pull request merged**